pennylane/transforms/draw.py

```diff
@@ -540,7 +552,13 @@ def circuit2(x, y):
 
     @wraps(qnode)
     def wrapper(*args, **kwargs_qnode):
-        qnode.construct(args, kwargs_qnode)
+        original_expansion_strategy = getattr(qnode, "expansion_strategy", None)
+
+        try:
+            qnode.expansion_strategy = expansion_strategy or original_expansion_strategy
+            qnode.construct(args, kwargs_qnode)
+        finally:
+            qnode.expansion_strategy = original_expansion_strategy

```

Comment:
I didn't know about `finally`. Cool! Is it necessary here or is this best practices?

Reply:
honestly, this is just copying what the old draw method did...